Health.py

```diff
@@ -1,39 +1,78 @@
-# For Healing:
-# vitC heals health
-# vitB heal magic
-# supplies repair mechanical
+FoodDict = {"Bread": [10, 0],
+            "Apple": [5, 0],
+            "Chicken Head": [20, 20],
+            "Thorny Fruit": [-20, -20]
+            }
 
-# food item would be ['foodName', 'food', [vitC_Value, vitB_Value]]
+RepairItems = {"Wrench": [10],
+               "Toolkit": [30],
+               "Repair Box": [100]
+            }
 
-# Enemy has health/magic/mechanical meters, attack & defense profile, Loot
-# If mechanical or health are zero, you cannot attack
-# If magic is zero, you may not use attacks with more than 5 magic damage
-# Defense profile is additive and subtracts from attack
-# Only one attack at a time, alternating turns
+PlayerStats = {"Health": 100,
+             "Magic": 100,
+             "Supply": 100}
 
-foodList = [["bread", 10, 0], ["apple", 5, 0], ["chicken head", 20, 20]]
-# potions = [['small red potion', 15], ['medium red potion', 30], ['big red potion', 50]]
 
-meterArray = [["health", 100], ["magic", 100], ["supply", 100]]
 
+def Heal(foodDict, meter):
+    print("Choose a healing item:")
+    for item, values in foodDict.items():
+        print(f"{item}: {values[0]} health points and {values[1]} Magic points")
+    for item, values in RepairItems.items():
+        print(f"{item}: {values[0]} supply points")
+    selectedItem = input('Item: ')
 
-def heal(
-    foodChoice, meter
-):  # searches food items for needed nutritional value then consumes them for points, changes health meter accordingly
+    if selectedItem in foodDict:
+        # Update the health and magic meters directly and store the updated values
+        meter["Health"] = max(0, min(100, meter["Health"] + foodDict[selectedItem][0]))
+        meter["Magic"] = max(0, min(100, meter["Magic"] + foodDict[selectedItem][1]))
```

Comment:
Are the health and magic meters percentages?

Is it possible that a consumable may increase magic and take away health and that's why we need to check if the meter is below 0?

Reply:
If we're adding the item to the meter, why would the limit got under 0?

Why can't the meter go over 100 if it's not a percentage? Why can't someone have 101 health?

Reply:
I assumed that we're implementing a game with a player progression structure of some sort and that usually implies that there's "level ups" included. 

So when a player levels up, their character attributes such as max health, max magic or etc, can be increased with attribute points gained from leveling up. "100 is the start default and can be further increased down the road" is what I was thinking. Meaning that the limiter can be changed as player progresses. 

Also, if the user did not get capped, it could cause unfair use of the functionality. For example, if the user stacked 10 golden apple consumables and then used them all in one go, he would be a tank and run through stages of the game that are at a higher difficulty level than what the current player progress is at.

Reply:
And to answer the first question, I had the idea in mind of some consumables being used berserker style. That means some items could take away some health and in return grant a bigger magic restoration (like -20 health and restore 50 magic/mana). 

But lets say the user's health is at 15 and he decided to use this berserker style item; it would be effectively suicide. So in this case, the user's health drops to -5 but having -5 health doesn't make sense because at 0 you're already dead. So unless there's some overkill function that we plan to implement, it makes no sense to go under zero, which is why the limit is there.

Reply:
Ah okay I see what you're getting at, and I am a big fan of that idea. Change it to
```python
meter["Health"] = max(0, min(MAX_HEALTH_VALUE, meter["Health"] + foodDict[selectedItem][0]))
meter["Magic"] = max(0, min(MAX_MAGIC_VALUE, meter["Magic"] + foodDict[selectedItem][1]))
```
and set those constants, then you'll have yourself an approved PR.